src/components/Drawer/Drawer.tsx

```diff
@@ -75,8 +76,8 @@ export const Drawer = ({
    */
   const [activeFocus, setActiveFocus] = useState(isActive);
   const windowRef = useRef<HTMLElement | null>(null);
-  const ref = useRef<HTMLDivElement | null>(null);
   const isMountedRef = useRef(true);
+  const onCloseRef = useRef(onClose);
```

Comment:
Capturing a ref to the `onClose` prop, so we can use it in callbacks and effects without worrying about whether the dep has changed or not.